crates/katana/runner/src/lib.rs

```diff
@@ -157,3 +188,94 @@ impl Drop for KatanaRunner {
         }
     }
 }
+
+type Provider = FillProvider<
+    JoinFill<
+        JoinFill<JoinFill<JoinFill<Identity, GasFiller>, NonceFiller>, ChainIdFiller>,
+        SignerFiller<EthereumSigner>,
+    >,
+    RootProvider<Http<Client>>,
+    Http<Client>,
+    Ethereum,
+>;
+
+#[derive(Debug)]
+pub struct AnvilRunner {
+    process: Child,
+    provider: Provider,
+    pub endpoint: String,
+}
+
+impl AnvilRunner {
+    pub async fn new() -> Result<Self> {
+        let port = find_free_port();
+
+        let process = Command::new("anvil")
+            .arg("--port")
+            .arg(&port.to_string())
+            .arg("--silent")
+            .arg("--disable-block-gas-limit")
+            .spawn()
+            .expect("Could not start background Anvil");
```

Comment:
@kariy I commented to @fabrobles92 that we prefer a runner over having to import Anvil and depend on it directly.
Is that ok for you, or do you prefer Anvil to be integrated? As we should migrate to alloy too though for the types.

Reply:
ah no, i mean i agree on the idea of the runner, but i think it's better to use the builder pattern that `Anvil` already provides to spawn the process instead of doing the `Command` chain manually, no? or is that what you're actually referring to?

Reply:
> ah no, i mean i agree on the idea of the runner, but i think it's better to use the builder pattern that `Anvil` already provides to spawn the process instead of doing the `Command` chain manually, no? or is that what you're actually referring to?

Yeah, I had in mind like Katana and only rely on the `Command` to decouple. Because alloy core that we should use during this re-work to send transactions doesn't contain `Anvil`, which may force to have both ethers and alloy, only for `Anvil`.